src/libs/ReportSecondaryActionUtils.ts

```diff
@@ -217,12 +217,25 @@ function isUnapproveAction(report: Report, policy?: Policy): boolean {
     const isReportApproved = isReportApprovedUtils({report});
     const isReportSettled = isSettled(report);
     const isPaymentProcessing = report.isWaitingOnBankAccount && report.statusNum === CONST.REPORT.STATUS_NUM.APPROVED;
+    const isAdmin = policy?.role === CONST.POLICY.ROLE.ADMIN;
 
     if (isReportSettled || isPaymentProcessing) {
         return false;
     }
 
-    return isExpenseReport && isReportApprover && isReportApproved;
+    if (!isExpenseReport || !isReportApproved) {
+        return false;
+    }
+
+    // Only final approver or admins can unapprove approved reports
+    if (report.statusNum === CONST.REPORT.STATUS_NUM.APPROVED) {
+        const finalApproverAccountID = report.managerID;
+        const isFinalApprover = finalApproverAccountID === getCurrentUserAccountID();
+        return isFinalApprover || isAdmin;
+    }
```

Comment:
![457584091-e4de1e4b-6fa7-4dd3-b241-3c131c1e1a47](https://github.com/user-attachments/assets/8aa16e64-6d70-46f1-9fbb-cba5d46302da)


```suggestion
    if (report.statusNum === CONST.REPORT.STATUS_NUM.APPROVED) {
        return isManager || isAdmin;
    }
```